lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -83,7 +84,7 @@ def host_name(java_opts, credentials)
 
       def node_name(java_opts, credentials)
         name = credentials['node-name'] || @configuration['default_node_name']
-        java_opts.add_system_property('appdynamics.agent.nodeName', "#{name}")
+        java_opts.add_system_property('appdynamics.agent.nodeName', @appdynamics_tier_name + ":#{name}")
```

Comment:
@youngm is suggesting a change, which I think is a good one, to add the application name to the default node name.  And he suggests doing this in [`config/app_dynamics_agent.yml`](https://github.com/cloudfoundry/java-buildpack/blob/master/config/app_dynamics_agent.yml#L21) which is the appropriate location.